common/src/java/org/apache/hadoop/hive/common/FileUtils.java

```diff
@@ -1376,6 +1376,12 @@ public static RemoteIterator<FileStatus> listStatusIterator(FileSystem fs, Path
         status -> filter.accept(status.getPath()));
   }
 
+  public static RemoteIterator<LocatedFileStatus> listLocatedStatusIterator(FileSystem fs, Path path, PathFilter filter)
```

Comment:
why do you need this, you could simply use `listStatusIterator` as in `getHdfsDirSnapshotsForCleaner`

Reply:
it's not required, FileStatus object is enough

Reply:
Changed code to use FileStatus object and using existing _org.apache.hadoop.hive.common.FileUtils#listStatusIterator()_ API which is used in _org.apache.hadoop.hive.ql.io.AcidUtils#getHdfsDirSnapshotsForCleaner()_ API.